examples/example-node-telemetry/README.md

```diff
@@ -0,0 +1,111 @@
+# Telemetry demo app using data ingest and Atlas Charts
+
+A [node.js](https://nodejs.org) application to demonstrate how to use [Atlas Device Sync](https://www.mongodb.com/atlas/app-services/device-sync) and [Realm JavaScript](https://github.com/realm/realm-js) to read sensor data, store the data in [Atlas](https://www.mongodb.com/atlas), and visualize it with [Charts](https://www.mongodb.com/products/charts).
+
+The file structure is as follows:
+
+```
+├── README.md
+├── package.json
+├── src
+│   ├── app.ts
+│   ├── config.ts
+│   └── models
+│       ├── machine_info.ts
+│       └── sensor_reading.ts
+```
+
+* `src/app.ts` - the actual application
+* `src/config.ts` - contains the configuration (app id)
+* `src/models/` - the model classes
+
+## Altas App Services
+
+To use the app together with Atlas App Services, you need to create an Atlas database and collection. Moreover, you need to create App service and enable Device Sync. 
+
+The schema to use (App Services / Data Access / Schema):
+
+```json
+{
+  "title": "SensorReading",
+  "type": "object",
+  "required": [
+    "_id",
+    "freemem",
+    "timestamp",
+    "uptime"
+  ],
+  "properties": {
+    "_id": {
+      "bsonType": "objectId"
+    },
+    "freemem": {
+      "bsonType": "long"
+    },
+    "loadAvg": {
+      "bsonType": "array",
+      "items": {
+        "bsonType": "float"
+      }
+    },
+    "machineInfo": {
+      "title": "MachineInfo",
+      "type": "object",
+      "required": [
+        "platform",
+        "release"
+      ],
+      "properties": {
+        "platform": {
+          "bsonType": "string"
+        },
+        "release": {
+          "bsonType": "string"
+        }
+      }
+    },
+    "timestamp": {
+      "bsonType": "date"
+    },
+    "uptime": {
+      "bsonType": "float"
+    }
+  }
+}
+```
+
+Data can be visualized by [Charts](https://www.mongodb.com/products/charts). An example is shown below.
+
+![An example on how Charts can visualize incoming data](./charts-example.png)
+
+
+## How to build and run
+
+You need to clone Realm JavaScript's git repository:
+
+```sh
+git clone https://github.com/realm/realm-js
+cd realm-js
+git submodule update --init --recursive
+```
+
+Moreover, you need to install the dependencies for this app:
+
+```sh
+cd examples/example-node-telemetry
+npm install
+```
+
+Before building the app, you need to add your app id to `src/config.ts`. After that, you can build and run the app:
+
+```sh
+npm run build
+node dist/app.js
+```
```

Comment:
One improvement I think we could do here to provide a better DX is to let the example be independent of our repo. So e.g. having the README speak from the perspective of this example directory as the root. (Since `realm` is defined in the `package.json`, they shouldn't need our repo.)

I think the lines I marked here could be replaced with the following if you agree (ps. temporarily placing `\` in front of code blocks otherwise I can't show it in one code block here):
```
### Install Dependencies

\```sh
npm install
\```

### Run the App

1. [Copy your Atlas App ID](https://www.mongodb.com/docs/atlas/app-services/reference/find-your-project-or-app-id/#std-label-find-your-app-id) from the App Services UI.
2. Paste the copied ID as the value of the existing variable `ATLAS_APP_ID` in [src/atlas-app-services/config.ts](./src/atlas-app-services/config.ts):
\```ts
export const ATLAS_APP_ID = "YOUR_APP_ID";
\```

3. Start the app

\```sh
npm run build
npm start
\```
```

Notes:
* The mention of `ATLAS_APP_ID` assumes a refactor of the code, but you can always change it along with the link if you decide not to refactor 👍
* The code shown under `Start the app` assumes that a `start` script has been added in `package.json` (I left a previous alternative where the `start` script also calls the `build` script. If so, you could just show `npm start`).

Reply:
I think it is related to https://github.com/realm/realm-js/pull/6116/files#r1321373935

Reply:
Yeah seems related 🙂 I believe the example runs fine without the `git submodule update`. Also if you e.g. copy the example directory to outside the js repo, it should run fine.

Reply:
True. And as @takameyer is saying, we should find a way to make the example easily accessible.

Reply:
If we just don't add this to the workspaces, then it will treat itself like a separate project within `realm-js`.  Currently it is not in workspaces, so I guess the `git submodule update` isn't even a necessary step here.  It would just download the `realm` package into the local `node_modules`

Reply:
Also, we should know as a team that if we want to test these examples against the local version of `realm`, this could be done by adding the example to the workspaces.  The only issue would be with the React Native projects, as these require a bit of configuration changes to work within the mono-repo.